File: payola/web/public/javascripts/cz/payola/web/client/views/graph/LocationDescriptor.js
goog.provide('cz.payola.web.client.views.graph.LocationDescriptor');
goog.require('cz.payola.web.client.views.Point');
goog.require('cz.payola.web.client.views.Vector');
cz.payola.web.client.views.graph.LocationDescriptor.informationPositionCorrection = new cz.payola.web.client.views.Vector(0.0, 4.0);
cz.payola.web.client.views.graph.LocationDescriptor.getVertexInformationPosition = function(position) {
var self = this;
return position.$plus(self.informationPositionCorrection);
};
cz.payola.web.client.views.graph.LocationDescriptor.getEdgeInformationPosition = function(originPosition, destinationPosition) {
var self = this;
var x = ((originPosition.x + destinationPosition.x) / 2);
var y = ((originPosition.y + destinationPosition.y) / 2);
return new cz.payola.web.client.views.Point(x, y).$plus(self.informationPositionCorrection);
};
cz.payola.web.client.views.graph.LocationDescriptor.metaClass_ = new s2js.MetaClass('cz.payola.web.client.views.graph.LocationDescriptor', []);

File: payola/web/client/src/main/scala/cz/payola/web/client/views/graph/LocationDescriptor.scala
package cz.payola.web.client.views.graph

import cz.payola.web.client.views.{Vector, Point}

object LocationDescriptor {

    private val informationPositionCorrection = Vector(0, 4)

    def getVertexInformationPosition(position: Point): Point = {
        position + informationPositionCorrection
    }
    
    def getEdgeInformationPosition(originPosition: Point, destinationPosition: Point): Point = {
        val x = (originPosition.x + destinationPosition.x)/2
        val y = (originPosition.y + destinationPosition.y)/2
        Point(x, y) + informationPositionCorrection
    }
}